.github/workflows/check-manifests-changes.yml

```diff
@@ -7,21 +7,38 @@ on:
   pull_request_target: { }
 
 env:
-  PR_CACHE_KEY: pr-manifests-${{ github.run_id }}-${{ github.run_attempt }}
-  MAIN_CACHE_KEY: main-manifests-${{ github.run_id }}-${{ github.run_attempt }}
+  PR_CACHE_KEY: pr-manifests-${{ github.event.pull_request.number }}-${{ github.event.pull_request.head.sha }}
+  MAIN_CACHE_KEY: main-manifests-${{ github.event.pull_request.number }}-${{ github.event.pull_request.base.sha }}
 
 jobs:
   create-pr-manifests:
     runs-on: ubuntu-latest
+    outputs:
+      cache-key: ${{ env.PR_CACHE_KEY }}
+      success: ${{ steps.create-manifests.outcome }}
     steps:
       - name: Checkout PR branch
         uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
 
       - name: Create manifests on PR branch
+        id: create-manifests
         run: |
-          make dry-run-control-plane
+          echo "Creating manifests for PR branch..."
+          if ! make dry-run-control-plane; then
+            echo "❌ Failed to create manifests on PR branch"
+            exit 1
+          fi
+
+          if [[ ! -f ./dry-run/manifests.yaml ]]; then
+            echo "❌ Manifest file not found after make command"
+            exit 1
+          fi
+
           mkdir -p ./cache/pr
-          mv ./dry-run/manifests.yaml ./cache/pr/manifests.yaml
+          cp ./dry-run/manifests.yaml ./cache/pr/manifests.yaml
```

Comment:
Why the change? Why `cp` instead of `mv`?

Reply:
To preserve the original:
The workflow pattern is:
- Generate manifest → ./dry-run/manifests.yaml
- Copy to job-specific cache → ./cache/pr/manifests.yaml or ./cache/main/manifests.yaml
- Cache the job-specific directory for use in the diff job